File: src/main/java/com/thebluehats/server/core/modules/CustomEnchantUtilsModule.java
package com.thebluehats.server.core.modules;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.google.inject.Singleton;
import com.thebluehats.server.game.managers.enchants.CustomEnchantUtils;
import com.thebluehats.server.game.utils.RomanNumeralConverter;

public class CustomEnchantUtilsModule extends AbstractModule {
    @Provides
    @Singleton
    static CustomEnchantUtils provideCustomEnchantUtils(RomanNumeralConverter romanNumeralConverter) {
        return new CustomEnchantUtils(romanNumeralConverter);
    }

    @Override
    protected void configure() {
    }
}

File: src/main/kotlin/com/thebluehats/server/core/modules/CustomEnchantUtilsModule.kt
package com.thebluehats.server.core.modules

import com.google.inject.AbstractModule
import com.google.inject.Provides
import com.google.inject.Singleton
import com.thebluehats.server.game.managers.enchants.CustomEnchantUtils

class CustomEnchantUtilsModule : AbstractModule() {
    override fun configure() {}

    companion object {
        @Provides
        @Singleton
        @JvmStatic
        fun provideCustomEnchantUtils(): CustomEnchantUtils {
            return CustomEnchantUtils()
        }
    }
}